internal/sequencerapi/api.go

```diff
@@ -0,0 +1,94 @@
+package sequencerapi
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/internal/ethapi"
+	"github.com/ethereum/go-ethereum/metrics"
+	"github.com/ethereum/go-ethereum/rpc"
+)
+
+var (
+	sendRawTxConditionalCostMeter       = metrics.NewRegisteredMeter("sequencer/sendRawTransactionConditional/cost", nil)
+	sendRawTxConditionalRequestsCounter = metrics.NewRegisteredCounter("sequencer/sendRawTransactionConditional/requests", nil)
+	sendRawTxConditionalAcceptedCounter = metrics.NewRegisteredCounter("sequencer/sendRawTransactionConditional/accepted", nil)
+)
+
+type sendRawTxCond struct {
+	b ethapi.Backend
+}
+
+func GetSendRawTxConditionalAPI(b ethapi.Backend) rpc.API {
+	return rpc.API{
+		Namespace: "eth",
+		Service:   &sendRawTxCond{b},
+	}
+}
+
+func (s *sendRawTxCond) SendRawTransactionConditional(ctx context.Context, txBytes hexutil.Bytes, cond types.TransactionConditional) (common.Hash, error) {
+	sendRawTxConditionalRequestsCounter.Inc(1)
+
+	cost := cond.Cost()
+	sendRawTxConditionalCostMeter.Mark(int64(cost))
+	if cost > types.TransactionConditionalMaxCost {
+		return common.Hash{}, &jsonRpcError{
+			message: fmt.Sprintf("conditional cost, %d, exceeded max: %d", cost, types.TransactionConditionalMaxCost),
+			code:    types.TransactionConditionalCostExceededMaxErrCode,
+		}
+	}
+
+	// Perform sanity validation prior to state lookups
+	if err := cond.Validate(); err != nil {
+		return common.Hash{}, &jsonRpcError{
+			message: fmt.Sprintf("failed conditional validation: %s", err),
+			code:    types.TransactionConditionalRejectedErrCode,
+		}
+	}
+
+	state, header, err := s.b.StateAndHeaderByNumber(context.Background(), rpc.LatestBlockNumber)
```

Comment:
If we use `rpc.LatestBlockNumber` here then it will create an incentive for MEV bots to only use this endpoint. Other implementations use state from a few blocks in the past, i recommend we do the same thing. We could be simple here and hardcode a random number like `tip - 3` or allow for it to be configurable

Reply:
So we already check the conditional against `tip` and `tip-1` (line 70 of this file) to disincentivize MEV on this endpoint. Open to making this depth configurable as well

Reply:
Why do we check twice instead of just checking once on the older state?

Reply:
ahh good catch, yea the state only needs to be checked on the parent. Adjusting that